### PR TITLE
KCL-10859 Get current iframe url from preview

### PR DESCRIPTION
**Pull request by PavelVyskocil**
### Motivation

Adds the ability to get the current iframe URL from preview (KCL-10859)
Related to the new feature `Multi-resolution preview in Web Spotlight` (KCL-10856)

### Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### CODE REVIEW

**Review by JanBarton — APPROVED** [review]

**Review by vladbulyukhin — CHANGES_REQUESTED** [review]

File: src/lib/IFrameCommunicatorTypes.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 interface ISDKSupportedFeatures {
   readonly refreshHandler: boolean;
+  readonly currentPreviewIFrameUrlHandler: boolean;
```
[the remark this answers is not in the document]
Comment (PavelVyskocil) [reply src/lib/IFrameCommunicatorTypes.ts]: renamed to `previewIFrameCurrentUrlHandler`

File: src/lib/IFrameCommunicatorTypes.ts [hunk 2]
```diff
@@ -45,6 +46,10 @@ export interface IRefreshMessageMetadata {
   readonly manualRefresh: boolean;
 }
 
+export interface IGetCurrentPreviewIFrameUrlMessageData {
```
[the remark this answers is not in the document]
Comment (PavelVyskocil) [reply src/lib/IFrameCommunicatorTypes.ts]: renamed to `IPreviewIFrameCurrentUrlMessageData`

File: src/sdk.ts [hunk 3]
```diff
@@ -158,6 +164,14 @@ class KontentSmartLinkSDK {
       reload();
     }
   };
+
+  private handleGetCurrentIFrameUrlMessage = (): void => {
+    const messageData: IGetCurrentPreviewIFrameUrlMessageData = {
+      previewUrl: window.self.location.href,
```
[the remark this answers is not in the document]
Comment (PavelVyskocil) [reply src/sdk.ts]: Fixed, thanks :)

File: src/lib/IFrameCommunicatorTypes.ts [hunk 4]
```diff
@@ -117,6 +122,8 @@ export enum IFrameMessageType {
   AddInitial = 'kontent-smart-link:add:initial',
   AddAction = 'kontent-smart-link:add:action',
   RefreshPreview = 'kontent-smart-link:preview:refresh',
+  PreviewIFrameCurrentUrl = 'kontent-smart-link:preview:currentUrl',
```
Comment (vladbulyukhin) [inline src/lib/IFrameCommunicatorTypes.ts:125]: Sorry, that I didn't notice it earlier, but `currentUrl` should actually be `current-url` here to be consistent with other event names where we follow this convention.

Comment (PavelVyskocil) [reply src/lib/IFrameCommunicatorTypes.ts]: fixed, thanks

**Review by vladbulyukhin — APPROVED** [review]

**Pull request merged**